movementlabsxyz/movement
# Correct move event monitoring.

# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.


Correct the movement event monitoring. 
# Changelog
Use Rest API and not the move client.
Update event processing and deserialization.
Add event process sequence number saving to avoid pulling them another time.
Save the event state in a file.


# Testing


- Start the suzuka node / bridge using process compose:
 ```
CELESTIA_LOG_LEVEL=FATAL CARGO_PROFILE=release CARGO_PROFILE_FLAGS=--release  nix develop --extra-experimental-features nix-command --extra-experimental-features flakes --command bash  -c "just bridge native build.setup.eth-local.celestia-local.bridge --keep-tui"
``` 
- Run the test  `test_movement_event` that sends an initiate transfer event:
```
DOT_MOVEMENT_PATH="$(pwd)/.movement" cargo test --test bridge_e2e_test test_movement_event -- --nocapture --test-threads=1 
```
- The event should be processed by the bridge in the bridge process compose logs.
# Outstanding issues
